src/commands/periscope/helpers/periscopehelper.ts

```diff
@@ -0,0 +1,258 @@
+                    return undefined;
+                }
+                storageAccountNameToStorageIdMap.set(name!, item.storageAccountId!);
+            }
+        });
+
+        // Create quick pick for more than 1 storage account scenario.
+        const selectedStorageAccount = await vscode.window.showQuickPick(
+            Array.from(storageAccountNameToStorageIdMap.keys()).map((label) => ({ label })),
+            {
+                placeHolder: "Select storage account for periscope deployment:",
+                ignoreFocusOut: true,
+                onDidSelectItem: (item) => { return item.toString(); }
+            });
+
+        if (selectedStorageAccount?.label) {
+            return storageAccountNameToStorageIdMap.get(selectedStorageAccount!.label);
+        } else {
+            return undefined;
+        }
+
+    } else if (diagnosticSettings && diagnosticSettings.value!.length === 1) {
+        // In case of only one storage account associated, use the one (1) as default storage account and no UI will be displayed.
+        const selectedStorageAccount = diagnosticSettings.value![0].storageAccountId!;
+        return selectedStorageAccount;
+    }
+
+    return undefined;
+}
+
+export async function getStorageInfo(
+    target: k8s.CloudExplorerV1.CloudExplorerResourceNode,
+    diagnosticStorageAccountId: string
+): Promise<PeriscopeStorage | undefined> {
+    try {
+        const clusterStorageInfo = <PeriscopeStorage>{};
+        const { resourceGroupName, name: accountName } = parseResource(diagnosticStorageAccountId!);
+
+        if (!resourceGroupName || !accountName) {
+            vscode.window.showErrorMessage(`Invalid storage id ${diagnosticStorageAccountId} associated with the cluster`);
+            return;
+        }
+
+        // Get keys from storage client.
+        const cloudResource = target.cloudResource;
+        const storageClient = new ast.StorageManagementClient(cloudResource.root.credentials, cloudResource.root.subscriptionId);
+        const storageAccKeyList = await storageClient.storageAccounts.listKeys(resourceGroupName, accountName);
+        clusterStorageInfo.storageName = accountName;
+        clusterStorageInfo.storageKey = storageAccKeyList.keys?.find((it) => it.keyName === "key1")?.value!;
+
+        // Generate 5 mins downlable shortlived sas along with 7 day shareable SAS.
+        clusterStorageInfo.storageDeploymentSas = getSASKey(accountName, clusterStorageInfo.storageKey!, SASExpiryTime.FiveMinutes);
+        clusterStorageInfo.sevenDaysSasyKey = getSASKey(accountName, clusterStorageInfo.storageKey!, SASExpiryTime.SevenDays);
+
+        return clusterStorageInfo;
+    } catch (e) {
+        vscode.window.showErrorMessage(`Storage associated with cluster had following error: ${e}`);
+        return undefined;
+    }
+}
+
+export function writeTempAKSDeploymentFile(file: any) {
```

Comment:
💡 This is actually is associated with another awesome idea (from what:  I think there is one associated comment above), so essentially, I will make this function much simpler and all the `finish` capture stuff moved here, so that once the chunks are done , then the `replacement` functionalists will be called.